### DISCUSSION TITLE
"Eye-Test" Calyx Sharing numbers for Lenet and Squeezenet

### DISCUSSION THREAD

**Discussion by calebmkim** [cucapra/futil#1182] [General]
The number of resources used in a fully inlined (and then compiled) Calyx file, with different sharing bounds. Since it was fully inlined, I copy and pasted just the main component into a separate file and ran ctrl+f for different resources to get the estimates. 

As a refresher for what the bounds mean, the first number is the number of times a combinational component will be shared, second number is the number of times a register will be shared, and third number is number of times all other components will be shared. I can check the numbers for sharing for combinational components too , but I think @rachitnigam mentioned that we probably don't want any sharing of these types of cells. Another disclaimer to add is that I haven't verified the correctness of these large Calyx files either.

Lenet: 
| (ctrl + f)  | no sharing | bounds = 1,2,2 | bounds=1,4,4 | always share |
| ------------- | ------------- | ------------- | ------------- | ------------- |
| "std_reg"  | 601  | 517  | 478  | 454  |
| "std_fp_smult_pipe" | 38 | 34  | 32  | 31  |
| "std_mult_pipe" | 8| 8  | 8  | 8  |
| "std_div_pipe" | 3  | 2  | 2  | 1  |

Squeezenet: 
| (ctrl + f)  | no sharing | bounds = 1,2,2 | bounds=1,4,4 | always share |
| ------------- | ------------- | ------------- | ------------- | ------------- |
| "std_reg"  | 2197  | 1774  | 1565  | 1428  |
| "std_fp_smult_pipe" | 27 | 14  | 7  | 1  |
| "std_mult_pipe" | 60 | 60  | 60  | 60  |
| "std_div_pipe" | 1  | 1  | 1 | 1  |

As you can see, there wasn't much sharing of multipliers (but the register sharing looks good, and you can see that the bounded sharing stuff is working as expected). I think if you guys agree with my thinking on #1181 and its corresponding issue (#1180) then I can retry the sharing numbers for multipliers.

**Comment by rachitnigam** [comment]
Wowowow! These numbers are already very cool. I am particularly excited to see `std_fp_smult_pipe` getting shared down to exactly `1` for Squeezenet!

@sampsyo even without synthesis results, this table is looking pretty compelling for a short paper. I think even if we can't figure out a way to get resource estimates, this seems a plausible direction for a short paper
